projects/reservoir/ks.py

```diff
@@ -0,0 +1,141 @@
+from dataclasses import dataclass
```

Comment:
In code that is directly copied from or heavily influenced by another source, it is customary to include a copy of its license as a comment at the top of the module.  See, e.g., [here in xarray](https://github.com/pydata/xarray/blob/ab726c536464fbf4d8878041f950d2b0ae09b862/xarray/coding/cftimeindex.py#L2-L40) or [here within fv3net](https://github.com/ai2cm/fv3net/blob/ff1e7b365f1553e03ae7a00862e426a3a3160631/external/vcm/vcm/cubedsphere/_skimage.py#L1-L37).